kuma/javascript/src/payments/subscription-form.jsx

```diff
@@ -0,0 +1,107 @@
+// @flow
+import * as React from 'react';
+import { useRef, useState } from 'react';
+
+import { getLocale, gettext, Interpolated } from '../l10n.js';
+
+export default function SubscriptionForm() {
+    const locale = getLocale();
+    const subscriptionFormRef = useRef(null);
+    const [subscribeFormDisabled, setSubscribeFormDisabled] = useState(false);
+    const [subscribeButtonEnabled, setSubscribeButtonEnabled] = useState(false);
+
+    const toggleButton = () => {
+        setSubscribeButtonEnabled(!subscribeButtonEnabled);
+    };
+
+    /**
+     * Opens Stripe modal allowing a user to complete their subscription.
+     * @param {Object} event - The form submit event
+     */
+    const submit = event => {
+        event.preventDefault();
+
+        const subscriptionForm = subscriptionFormRef.current;
+        if (subscriptionForm) {
+            const formData = new FormData(subscriptionForm);
+            setSubscribeFormDisabled(true);
+
+            const stripeHandler = window.StripeCheckout.configure({
+                key: window.mdn.stripePublicKey,
+                locale: 'auto',
+                name: 'MDN Web Docs',
+                zipCode: true,
+                currency: 'usd',
+                amount: 500,
+                email: '',
+                token: function(response) {
+                    formData.set('stripe_token', response.id);
+                    subscriptionForm.submit();
```

Comment:
> Yes.. no.. ish.. 😁 The flow is: stripe-client-modal gives us a token, we post that to our backend, which it uses to create a customer and a subscription via the Stripe API. Then our backend responds to our frontend with the new user data. So this form will need post to our backend ultimately and after that we could redirect (or we do it optimistically while submission is in progress).

That is what is happening here correct? So the Python code that recieves this form submission is the piece that needs to do the redirect to the Thank you page.

Reply:
Yup, we can do it like that!